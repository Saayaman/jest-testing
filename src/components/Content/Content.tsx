import React from 'react';

class Content extends React.Component<{name: string}> {
  render() {
    return (
      <div>Hello, {this.props.name}</div> 
    )
  }
}

export default Content;