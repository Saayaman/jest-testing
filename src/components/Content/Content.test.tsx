import React from 'react';
import ReactDOM from 'react-dom';
import Content from './Content';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe("content tests", () => {
  it("Content should render correctly", () => {
    const { getByText } = render(<Content name="Ayako" />)
    expect(getByText('Hello, Ayako')).toBeInTheDocument();
  });

  it("Content shallow render test", () => {
    const wrapper = shallow(<Content name="Erika" />);
    const welcome = <div>Hello, Erika</div> ;
    // expect(wrapper.contains(welcome)).toBe(true);
    expect(wrapper.contains(welcome)).toEqual(true);
  })

  it("Snapshottesting", () => {
    const tree = renderer.create(<Content name="Eugene" />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
