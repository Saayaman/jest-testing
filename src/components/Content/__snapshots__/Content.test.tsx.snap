// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content tests Snapshottesting 1`] = `
<div>
  Hello, 
  Eugene
</div>
`;
